napari_plugin_manager/qt_package_installer.py

```diff
@@ -374,27 +384,44 @@ def cancel(self, job_id: Optional[JobId] = None):
             Job ID to cancel.  If not provided, cancel all jobs.
         """
         if job_id is None:
+            all_pkgs = []
+            for item in deque(self._queue):
+                all_pkgs.extend(item.pkgs)
+
             # cancel all jobs
             self._queue.clear()
             self._end_process()
+            self.processFinished.emit(
+                1, 0, InstallerActions.CANCEL_ALL, all_pkgs
+            )
             return
 
-        for i, item in enumerate(self._queue):
+        for i, item in enumerate(deque(self._queue)):
             if item.ident == job_id:
                 if i == 0:  # first in queue, currently running
+                    self._queue.remove(item)
                     self._end_process()
                 else:  # still pending, just remove from queue
                     self._queue.remove(item)
+
+                self.processFinished.emit(
+                    1, 0, InstallerActions.CANCEL, item.pkgs
```

Comment:
```suggestion
                    {"exit_code": 1, "status_code": 0, "action": InstallerActions.CANCEL, "packages": item.pkgs}
```

Tiny nit here but for better readability maybe something like this?

Reply:
Fixed